internal/gateway/gateway.go

```diff
@@ -348,5 +353,11 @@ func (g *Gateway) Start() error {
 	<-terminationSignal
 	log.Println("Termination signal received. Shutting down...")
 
+	ctx, cancel := context.WithTimeout(g.ctx, g.Config.GatewayService.ShutdownTimeout)
+	defer cancel()
+	g.httpServer.Shutdown(ctx)
+	g.grpcServer.GracefulStop()
```

Comment:
https://github.com/grpc/grpc-go/issues/5930

Isn't this bad?

Reply:
Should we use .Stop instead?

Reply:
It looks like this was fixed in https://github.com/grpc/grpc-go/releases/tag/v1.53.0. We're on v1.62.0.